File: englising-data/model/track.py
from sqlalchemy import Column, ForeignKey, Integer, String, Float, Boolean, TIMESTAMP, DECIMAL, Text, Date, Table
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from model.base import Base


class Track(Base):
    __tablename__ = 'track'
    track_id = Column(Integer, primary_key=True)
    album_id = Column(Integer, ForeignKey('album.album_id'), nullable=False)
    track_index = Column(Integer, nullable=False)
    title = Column(String(255), nullable=False)
    spotify_id = Column(String(255), nullable=False)
    youtube_id = Column(String(255), nullable=False)
    isrc = Column(String(255), nullable=False)
    spotify_popularity = Column(Integer, nullable=False)
    duration_ms = Column(Integer, nullable=False)
    is_lyrics = Column(Boolean, nullable=False, default=False)
    feature_acousticness = Column(Float, nullable=True)
    feature_danceability = Column(Float, nullable=True)
    feature_energy = Column(Float, nullable=True)
    feature_tempo = Column(Float, nullable=True)
    feature_positiveness = Column(Float, nullable=True)
    created_at = Column(TIMESTAMP, nullable=False, default=func.now())
    updated_at = Column(TIMESTAMP, onupdate=func.now())
    album = relationship("Album", back_populates="tracks")
    artists = relationship("ArtistTrack", back_populates="track")

File: englising-be/src/main/java/org/englising/com/englisingbe/track/entity/Track.java
package org.englising.com.englisingbe.track.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.englising.com.englisingbe.album.entity.Album;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "track")
public class Track {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "track_id")
    private Long trackId;

    @ManyToOne
    @JoinColumn(name = "album_id", referencedColumnName = "album_id")
    private Album album;

    @Column(name = "track_index")
    private Integer trackIndex;

    @Column(name = "title")
    private String title;

    @Column(name = "spotify_id")
    private String spotifyId;

    @Column(name = "youtube_id")
    private String youtubeId;

    @Column(name = "isrc")
    private String isrc;

    @Column(name = "spotify_popularity")
    private String spotifyPopularity;

    @Column(name = "duration_ms")
    private Integer durationMs;

    @Column(name = "feature_acousticness")
    private Float featureAcousticness;

    @Column(name = "feature_danceability")
    private Float featureDancebility;

    @Column(name = "feature_energy")
    private Float featureEnergy;

    @Column(name = "feature_positiveness")
    private Float featurePositiveness;

    @Column(name = "genre")
    private String genre;

    @Column(name = "lyric_status")
    private String lyricStatus;

    @CreationTimestamp
    @Column(name = "created_at")
    private Timestamp createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Timestamp updatedAt;
}
